internal/walk/walklinks.go

```diff
@@ -20,8 +21,22 @@ func pathRewriter(src string, dest string, fn fs.WalkDirFunc) fs.WalkDirFunc {
 	}
 }
 
+func makeRegexp(str string) (re *regexp.Regexp, err error) {
+	if strings.HasPrefix(str, "/") {
+		str = `^` + str
+	}
+	if strings.HasSuffix(str, "/") {
+		str = strings.TrimRight(str, "/") + `\z`
+	}
+	str = strings.Replace(str, "**", `.*`, -1)
+	str = strings.Replace(str, "*", `[^/]+`, -1)
+	str = strings.Replace(str, "?", `[^/]*`, -1)
```

Comment:
This isn't right as ? matches exactly one character, but `[^/]*` matches any number of characters.  Should it be just `[^/]` ?

Reply:
That's right. I misread the rsync man page. Fixed.